packages/flutter/lib/src/painting/placeholder_span.dart

```diff
@@ -15,12 +15,16 @@ import 'text_style.dart';
 /// An immutable placeholder that is embedded inline within text.
 ///
 /// [PlaceholderSpan] represents a placeholder that acts as a stand-in for other
-/// content. A [PlaceholderSpan] by itself does not contain useful
-/// information to change a [TextSpan]. Instead, this class must be extended
-/// to define contents.
+/// content. A [PlaceholderSpan] by itself does not contain useful information
+/// to change a [TextSpan]. [WidgetSpan] from the widgets library extends
+/// [PlaceholderSpan] and may be used instead to specify a widget as the contents
+/// of the placeholder.
+///
+/// Flutter widgets such as [TextField], [Text] and [RichText] do not recognize
+/// [PlaceholderSpan] subclasses other than [WidgetSpan]. **Consider
```

Comment:
`RenderEditable` and `RenderParagraph` need to be moved to the widget library if we want to remove/deprecate `PlaceholderSpan`. Let me see if I can just `hide` `PlaceholderSapn`.

Reply:
That worked. @Hixie would it be ok to not exposing `PlaceholderSpan` from the painting library, since it's not supposed to be directly implemented (I think)?

Reply:
Oh wait with the ParentDataWidget trick maybe we don't need the PlaceholderSpan class anymore.

Reply:
ah I see. I guess that is why the PlaceholderSpan was first introduced. I think we should avoid package level private.

we could use `interface` modifier, but people can still implements it.

Should it still be considered valid to extends PlaceHolderSpan if developer builds on top Rendering library without Widget library?

Reply:
> Should it still be considered valid to extends PlaceHolderSpan if developer builds on top Rendering library without Widget library?

Yes but it would be very confusing when people write `extends PlaceholderSpan`, use the class in `Text` or `TextField`, and see their PlaceholderSpan class isn't working, and the inline widgets are all messed up. Like you said we could do a typedef and I think everything would work as usual?